eth/handler.go

```diff
@@ -452,9 +455,18 @@ func (pm *ProtocolManager) BroadcastBlock(block *types.Block, propagate bool) {
 
 	// If propagation is requested, send to a subset of the peer
 	if propagate {
+		// Calculate the TD of the block (it's not imported yet, so block.Td is not valid)
+		var td *big.Int
+		if parent := pm.chainman.GetBlock(block.ParentHash()); parent != nil {
+			td = new(big.Int).Add(parent.Td, block.Difficulty())
+		} else {
+			glog.V(logger.Error).Infof("propagating dangling block #%d [%x]", block.NumberU64(), hash[:4])
```

Comment:
Say two new blocks are mined within a very short time, but the miners knew about the first block, so the second is built on the first (not stale). If the second one is propagated further on the network before the first one, and we get it, then we cannot find the parent, and so will re-propagate it with 0 TD.

Can this have any unforeseen negative consequences on block propagation?